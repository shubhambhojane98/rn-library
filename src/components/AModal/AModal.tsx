import React from 'react';
import {StyleSheet, View, TouchableOpacity} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import IconSVG from '../../assets/svgs';
import {defaultScale} from '../../utils/Common';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';
import Modal from 'react-native-modal';

interface Props {
  top?: number;
  left?: number;
  right?: number;
  bottom?: number;
  backgroundColor?: string;
  children: any;
  closeButton?: boolean;
  visible?: boolean;
  borderRadius?: number;
  onDismiss?: any;
  animationIn?: any;
  animationOut?: any;
  style?: any;
  backdropColor?: string;
  backdropOpacity?: number;
  hasBackdrop?: boolean;
  theme: Theme;
}

const defaultMargin = 20;

const AModal: React.FC<Props> = ({
  backgroundColor,
  children,
  closeButton,
  left = defaultMargin,
  right = defaultMargin,
  top = defaultMargin,
  bottom = defaultMargin,
  visible,
  borderRadius,
  onDismiss,
  animationIn,
  animationOut,
  style,
  backdropColor,
  backdropOpacity,
  hasBackdrop,
}) => {
  const {colors} = useTheme();
  const stylesWithProp = styles({
    top,
    right,
    bottom,
    left,
  });
  return (
    <Modal
      animationIn={animationIn ? animationIn : 'slideInUp'}
      animationOut={animationOut ? animationOut : 'slideOutDown'}
      style={style}
      backdropColor={backdropColor}
      backdropOpacity={backdropOpacity}
      hideModalContentWhileAnimating={true}
      hasBackdrop={hasBackdrop}
      onBackdropPress={() => {
        onDismiss();
      }}
      isVisible={visible}>
      <View
        style={{
          ...stylesWithProp.modalView,
          backgroundColor: backgroundColor || colors.background,
          borderRadius,
        }}>
        {closeButton ? (
          <TouchableOpacity
            onPress={() => {
              onDismiss();
            }}
            style={stylesWithProp.CloseBtn}>
            <IconSVG name={'close'} height={20} width={20} />
          </TouchableOpacity>
        ) : null}
        {children}
      </View>
    </Modal>
  );
};
const styles = (props: {
  top: number;
  right: number;
  bottom: number;
  left: number;
}) =>
  StyleSheet.create({
    modalView: {
      paddingTop: moderateScale(40, defaultScale),
      position: 'absolute',
      borderRadius: moderateScale(16, 0.1),
      alignItems: 'center',
      top: moderateScale(props.top || defaultMargin, defaultScale),
      right: moderateScale(props.right || defaultMargin, defaultScale),
      bottom: moderateScale(props.bottom || defaultMargin, defaultScale),
      left: moderateScale(props.left || defaultMargin, defaultScale),
    },
    CloseBtn: {
      right: 0,
      position: 'absolute',
      padding: moderateScale(20, defaultScale),
    },
    modalOverlay: {
      position: 'absolute',
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
    },
  });
export default withTheme(AModal);
