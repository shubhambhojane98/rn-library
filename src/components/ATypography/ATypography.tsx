import React from 'react';
import {StyleProp, StyleSheet, Text, TextProps, TextStyle} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {fontFamilies} from '../../constants/fontFamilies';
import {TextAlignment, TypographyVariant} from './ATypographyEnum';
import {defaultScale} from '../../utils/Common';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';
interface OwnProps {
  variant?: TypographyVariant;
  color?: string;
  style?: StyleProp<TextStyle>;
  textAlign?: TextAlignment;
  fontSize?: number;
  theme: Theme;
}

const defaultFontSize = moderateScale(14, defaultScale);

type Props = OwnProps & TextProps;
const Typography: React.FC<Props> = ({
  variant = TypographyVariant.PRIMARY,
  color,
  style,
  textAlign = TextAlignment.Left,
  children,
  fontSize = defaultFontSize,
  ...otherProps
}) => {
  const {colors} = useTheme();
  return (
    <Text
      style={[
        styles({
          fontSize,
        })[variant],
        {color: color ? color : colors.textColor},
        {textAlign},
        style,
      ]}
      {...otherProps}>
      {children}
    </Text>
  );
};

export default withTheme(Typography);

const styles = (props: {fontSize: number}) =>
  StyleSheet.create({
    primary: {
      fontFamily: `${fontFamilies.Fonts.primary}-Regular`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'normal',
    },
    primarySemiBold: {
      fontFamily: `${fontFamilies.Fonts.primary}-SemiBold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: '600',
    },
    primaryBold: {
      fontFamily: `${fontFamilies.Fonts.primary}-Bold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'bold',
    },
    secondary: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Regular`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'normal',
    },
    secondaryDemi: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Demi`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: '600',
    },
    secondaryBold: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Bold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'bold',
    },
  });
