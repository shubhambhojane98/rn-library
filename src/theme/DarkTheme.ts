import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
