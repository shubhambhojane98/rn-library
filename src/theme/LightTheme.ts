import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
