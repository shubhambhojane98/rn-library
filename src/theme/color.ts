import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
