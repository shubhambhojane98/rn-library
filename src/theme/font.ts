import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }