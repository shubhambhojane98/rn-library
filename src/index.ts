import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
