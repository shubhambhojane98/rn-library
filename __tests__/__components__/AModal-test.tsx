import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AModal from '../../src/components/AModal/AModal';
import ATypography from '../../src/components/ATypography/ATypography';
import {TypographyVariant} from '../../src/components/ATypography/ATypographyEnum';
import {Color} from '../../src/theme';

it('renders correctly with default props', () => {
  const tree = renderer
    .create(
      <AModal
        visible={true}
        onDismiss={jest.fn()}
        closeButton={true}
        backgroundColor={Color.white}
        borderRadius={8}
        animationType={'fade'}
        children={
          <>
            <ATypography variant={TypographyVariant.PRIMARY_BOLD}>
              {' '}
              Modal content!
            </ATypography>
          </>
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with transparent', () => {
  const tree = renderer
    .create(
      <AModal
        visible={true}
        onDismiss={jest.fn()}
        closeButton={true}
        backgroundColor={Color.white}
        borderRadius={8}
        transparentOverlay={true}
        children={
          <>
            <ATypography variant={TypographyVariant.PRIMARY_BOLD}>
              {' '}
              Modal content!
            </ATypography>
          </>
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with closeButton hidden', () => {
  const tree = renderer
    .create(
      <AModal
        visible={true}
        onDismiss={jest.fn()}
        backgroundColor={Color.white}
        borderRadius={8}
        closeButton={false}
        children={
          <>
            <ATypography variant={TypographyVariant.PRIMARY_BOLD}>
              {' '}
              Modal content!
            </ATypography>
          </>
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
