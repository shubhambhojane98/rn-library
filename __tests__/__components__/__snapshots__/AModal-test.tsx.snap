// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with closeButton hidden 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.4)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "bottom": 24.57142857142857,
        "left": 24.57142857142857,
        "paddingTop": 49.14285714285714,
        "position": "absolute",
        "right": 24.57142857142857,
        "top": 24.57142857142857,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 21.13142857142857,
            "fontWeight": "bold",
          },
          Object {
            "color": "#231F20",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
       
      Modal content!
    </Text>
  </View>
</Modal>
`;

exports[`renders correctly with default props 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.4)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "bottom": 24.57142857142857,
        "left": 24.57142857142857,
        "paddingTop": 49.14285714285714,
        "position": "absolute",
        "right": 24.57142857142857,
        "top": 24.57142857142857,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 24.57142857142857,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <SvgMock
        height={20}
        name="close"
        width={20}
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 21.13142857142857,
            "fontWeight": "bold",
          },
          Object {
            "color": "#231F20",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
       
      Modal content!
    </Text>
  </View>
</Modal>
`;

exports[`renders correctly with transparent 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "bottom": 24.57142857142857,
        "left": 24.57142857142857,
        "paddingTop": 49.14285714285714,
        "position": "absolute",
        "right": 24.57142857142857,
        "top": 24.57142857142857,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 24.57142857142857,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <SvgMock
        height={20}
        name="close"
        width={20}
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 21.13142857142857,
            "fontWeight": "bold",
          },
          Object {
            "color": "#231F20",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
       
      Modal content!
    </Text>
  </View>
</Modal>
`;
